flash/image/classification/backbones/clip.py

```diff
@@ -0,0 +1,54 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from functools import partial
+
+import torch
+from torch import nn
+
+from flash.core.registry import FlashRegistry
+from flash.core.utilities.providers import _CLIP
+from flash.core.utilities.url_error import catch_url_error
+
+# Paper: Learning Transferable Visual Models From Natural Language Supervision
+# https://arxiv.org/abs/2103.00020 from Alec Radford et. al. (26 Feb 2021)
+# weights from https://github.com/openai/CLIP
+
+
+_CLIP_MODELS = {
+    "RN50": "resnet50",
+    "RN101": "resnet101",
+    "ViT_B_32": "vitb32",
+    "ViT_B_16": "vitb16",
+    "ViT_L_14": "vitl14",
+}
```

Comment:
Is there a reason not to use all available models in CLIP?
```
>>> import clip
>>> clip.available_models()
['RN50', 'RN101', 'RN50x4', 'RN50x16', 'RN50x64', 
'ViT-B/32', 'ViT-B/16', 'ViT-L/14', 'ViT-L/14@336px']
```

Reply:
Good catch! No reason not to have them :smiley: Added!